File: 백준/Silver/1629. 곱셈/곱셈.py
def modular(A, B, C):
    if B == 1:
        return A % C
    
    ret = modular(A, B // 2, C)
    ret = (ret * ret) % C

    if (B % 2 == 1):
        ret = (ret * A) % C
    
    return ret

if __name__ == "__main__":
    A, B, C = map(int, input().split())

    print(modular(A, B, C))

File: 백준/Silver/1629. 곱셈/곱셈.cc
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

ll A, B, C;

ll modular(ll a, ll b) {
    if (b == 1) return a % C;

    ll ret = modular(a, b/2);
    ret = (ret * ret) % C;

    if (b % 2) ret = (ret * a) % C;

    return ret;
}

int main() {
    scanf("%lld %lld %lld", &A, &B, &C);
    cout << modular(A, B) << '\n';
    return 0;
}